andela/balder-ah-backend
# #163457658 Article(s) should return boolean value for bookmarked property

#### What does this PR do?
This PR modifies the article controller to return a boolean value to know if a user has bookmarked the article or not


#### Description of Task to be completed?
 - Modify controller to get a single article to have a `bookmarked` property
 - Modify controller to get all articles to have a `bookmarked` property

#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Send a GET request to **localhost:3000/api/articles** to retrieve all the articles or **localhost:3000/api/articles/:slug** to retrieve a single article
 - The articles should have the `bookmarked` property
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**



#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories? (If applicable)

[#163457658](https://www.pivotaltracker.com/story/show/163457658)

#### Screenshots (If applicable)

N/A

#### Questions:
N/A
